documentcloud/entities/models.py

```diff
@@ -0,0 +1,75 @@
+from django.db import models
+from documentcloud.core.fields import AutoCreatedField, AutoLastModifiedField
+from wikidata.client import Client
+from django.utils.translation import gettext_lazy as _
+from .choices import EntityAccess
+
+
+class Entity(models.Model):
+    wd_entity = None
+
+    # A dictionary with language codes as keys.
+    name = models.CharField(max_length=500)
+    localized_names = models.JSONField()
+    wikidata_id = models.CharField(max_length=16)
+    # A dictionary with language codes as keys.
+    wikipedia_url = models.JSONField()
+    owner = models.ForeignKey(
+        "users.User", related_name="entities", on_delete=models.CASCADE
+    )
+    description = models.JSONField()
+    created_at = AutoCreatedField(
+        _("created at"),
+        db_index=True,
+        help_text=_("Timestamp of when the entity was created"),
+    )
+    updated_at = AutoLastModifiedField(
+        _("updated at"), help_text=_("Timestamp of when the entitywas last updated")
+    )
+    access = models.IntegerField(
+        _("access"),
+        choices=EntityAccess.choices,
+        help_text=_("Designates who may access this entity."),
+    )
+
+    def save(self, *args, **kwargs):
+        if not self.wikidata_id:
+            # Unnecessary?
+            raise "Missing wikidata_id in entity."
+
+        if not self.wikipedia_url:
+            self.wikipedia_url = self.get_url_for_wikidata_id()
+
+        if not self.name or not self.localized_names:
+            self.localized_names = self.get_names_for_wikidata_id()
+            # English bias here. TODO: How can this be addressed?
+            self.name = self.localized_names["en"]
+            if not self.name:
+                keys = self.localized_names.keys()
+                if len(keys) > 0:
+                    self.name = self.localized_names[keys[0]]
+                else:
+                    self.name = "Unknown"
```

Comment:
```
            # use .get if you are not sure it is in the dictionary
            self.name = self.localized_names.get("en")
            if not self.name:
                # the dictionary itself is falsey if empty
                if self.localized_names:
                     # can grab values directly, since you do not care about the key
                     # need to wrap in list as the default keys/values structure is not indexable
                     self.name = list(self.localized_names.values())[0]
                else:
                    # maybe we should log if this happens?  hopefully everything has a name in at least one language?
                    self.name = "Unknown"

Reply:
Ahh, nice. That's handy.